### ISSUE TITLE
trycloudflare CURLEWRITEERROR

### ISSUE THREAD

**Issue by Erquint** [LostRuins/koboldcpp#1155]
**Describe the Issue**
```
Load Text Model OK: True
Embedded KoboldAI Lite loaded.
Embedded API docs loaded.
Downloading Cloudflare Tunnel for Windows...
Remote Tunnel Failed!
Command 'curl -fL https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-windows-amd64.exe -o cloudflared.exe' returned non-zero exit status 23.
```

**Additional Information:**
Windows 10x64.
KCPP 1.75.2.
`koboldcpp_cu12.exe` executable ran from an unrestricted path.
Curl exit code [23 CURLEWRITEERROR](<https://curl.se/libcurl/c/libcurl-errors.html#CURLEWRITEERROR>).
I'd mention the launch flags if KCPP printed them to the terminal upon launching from GUI.

**Comment by LostRuins** [comment]
Using the tunnel option means it needs to download a Cloudflare tunnel binary. The error says this failed, check if your directory is read only.

**Comment by Erquint** [comment]
I don't think that's it.  
Which directory is it operating `curl` in?  
The one in which `koboldcpp_cu12.exe` is located or do you need me to unpack it..?  
![image](https://github.com/user-attachments/assets/11fdddb2-29ff-4512-a4aa-6002b6893e1e)

**Comment by Erquint** [comment]
Is Cygwin, MSYS2, Git Bash or something of that ilk used for curl?

**Comment by LostRuins** [comment]
curl is a standalone binary. It should come with every windows install.

You can test it in the command prompt with `curl example.com`

**Comment by Erquint** [comment]
Yes, I'm aware. I use it.  
It would help my own troubleshooting if you would tell me if Cygwin, MSYS2, Git Bash or something of that ilk used by koboldcpp. I think I spotted one of them acting weird around curl recently on a random occasion…

**Comment by LostRuins** [comment]
Nope, only basic curl is used. You can run it in a python debugger and step through the source, nothing special is done.
